django/db/backends/base/features.py

```diff
@@ -152,14 +152,13 @@ class BaseDatabaseFeatures:
     # Can the backend introspect a TimeField, instead of a DateTimeField?
     can_introspect_time_field = True
 
-    # Some backends may not be able to differentiate BigAutoField or
-    # SmallAutoField from other fields such as AutoField.
-    introspected_big_auto_field_type = 'BigAutoField'
-    introspected_small_auto_field_type = 'SmallAutoField'
-
-    # Some backends may not be able to differentiate BooleanField from other
-    # fields such as IntegerField.
-    introspected_boolean_field_type = 'BooleanField'
+    # Map fields which some backends may not be able to differentiate to the
+    # field it's introspected as.
+    introspected_field_types = {
```

Comment:
I would leave it empty (`introspected_field_types = {}`) by default and use `.get()` in tests, e.g.
```python
connection.features.introspected_field_types.get('BooleanField') or 'BooleanField'`
```

Reply:
I was just about to suggest this too :+1:

Reply:
I don't like this suggestion. What do you see as the advantage? Here's why I'm concerned:
- It's more verbose
- typos in the key name are silently ignored
- by implementing a mapping in the base features, it shows which fields are implemented in Django's tests (I don't think we should try to implement all fields proactively but accept PRs if a third party backends needs it)

Reply:
Agree with @timgraham

Reply:
OK, agreed.